### DISCUSSION TITLE
Additional "dev" dependencies in standalone output node_modules

### DISCUSSION THREAD

**Discussion by festusyuma** [vercel/next.js#59127] [Help]
### Summary

The standalone output of my apps were around 100Mb and It seemed a bit larger than expected. on further investigation. I noticed a couple of packages in the node_modules that seem to be dev dependencies. I deleted them and build output was dropped to 46MB. My questions are, 

- Why are these packages included?
- Are they being used somewhere I don't know of
- If they are not being used, is there anyhing I can do to exclude them from the node_modules in the build output

The folders I removed from the node_modules are
- @esbuild 9.3 MB
- @swc 35.9 MB

This is my next config
``` 
//@ts-check

// eslint-disable-next-line @typescript-eslint/no-var-requires
const { composePlugins, withNx } = require('@nx/next');

/**
 * @type {import('@nx/next/plugins/with-nx').WithNxOptions}
 **/
const nextConfig = {
  nx: {
    // Set this to true if you would like to use SVGR
    // See: https://github.com/gregberge/svgr
    svgr: false,
  },
  output: 'standalone',
  trailingSlash: true,
  images: {
    loader: 'cloudinary',
    path: 'https://res.cloudinary.com/dm1ta9lno/image/upload/',
  },
};

const plugins = [
  // Add more Next.js plugins to this list if needed.
  withNx,
];

module.exports = composePlugins(...plugins)(nextConfig);

```

NB: I am using nx monorepo

**Comment by ejasarevic** [comment]
Still valid. We also noticed many additional dependencies in standalone output `node_modules` which are not needed for the app itself.

**Comment by junlarsen** [comment]
I'm having JSDom and Vitest among others being output into chunks in standalone mode which makes no sense. It's confusing that there is zero documentation on how this process works, or how I can get this out of my bundle

**Comment by festusyuma** [comment]
Just realized I never added this here, there is an option to exclude files from the build via the next.config file, these are the current packages I need to exclude from being included in the node_modules

```
outputFileTracingExcludes: {
    '*': ['**@swc/core**', '**@esbuild/**', '**@rspack/**', '**@img/**'],
  },
```
